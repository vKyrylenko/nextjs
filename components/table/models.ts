import { SyntheticEvent } from 'react';

export interface IRowData {
  [key: string]: any;
}

export interface ITable {
  data?: Array<IRowData>;
  defaultSortColumn: string;
  cells: Array<ICell>;
  onRemoveItem?: (ids: string[]) => void;
  onExportItems?: (ids: string[]) => void;
  onToggleAddItem?: () => void;
  onSearch?: (search: string) => void;
}

export interface IRow {
  cells: Array<ICell>;
  keyName: string;
  onClick: (e, keyName) => void;
  selected?: boolean;
  isSelectable?: boolean;
  row: IRowData;
}

export interface ICell {
  name: string;
  width?: number;
  isSortable?: boolean;
  cellRenderer?: (data: IRowData, cellName: string) => void;
}

export interface ITableHead {
  onSelectAllClick: (e: SyntheticEvent) => void;
  order: 'asc' | 'desc';
  orderBy: string;
  cells: Array<ICell>;
  numSelected: number;
  rowCount: number;
  onRequestSort: (data: string) => void;
}

export interface ITableToolBar {
  numSelected?: number;
  onRemoveButtonClick: () => void;
  onExportButtonClick: () => void;
  onSearch?: (search: string) => void;
  onToggleAddButtonClick?: () => void;
}
