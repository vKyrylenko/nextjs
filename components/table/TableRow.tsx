import { FunctionComponent } from 'react';
import TableCell from '@material-ui/core/TableCell/TableCell';
import Checkbox from '@material-ui/core/Checkbox/Checkbox';
import MuiTableRow from '@material-ui/core/TableRow';
import makeStyles from '@material-ui/core/styles/makeStyles';
import { IRow } from './models';
import Link from 'next/link';

const useStyles = makeStyles(() => ({
  cell: {
    padding: '8px',
  },
  visuallyHidden: {
    border: 0,
    clip: 'rect(0 0 0 0)',
    height: 1,
    margin: -1,
    overflow: 'hidden',
    padding: 0,
    position: 'absolute',
    top: 20,
    width: 1,
  },
}));

export const TableRow: FunctionComponent<IRow> = ({ cells, onClick, row, isSelectable, keyName, selected }) => {
  const classes = useStyles();

  return (
    <MuiTableRow hover role="checkbox" aria-checked={selected} tabIndex={-1} key={keyName} selected={selected}>
      {isSelectable && (
        <TableCell padding="checkbox" onClick={(event) => onClick(event, keyName)}>
          <Checkbox checked={selected} />
        </TableCell>
      )}
      {cells.map((cell, index) => {
        return (
          <TableCell key={index} scope="row" className={classes.cell}>
            {cell.cellRenderer ? cell.cellRenderer(row, cell.name) : row[cell.name]}
          </TableCell>
        );
      })}
    </MuiTableRow>
  );
};
