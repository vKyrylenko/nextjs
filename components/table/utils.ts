export const SortOrders: { [index: string]: 'asc' | 'desc' } = {
  DESC: 'desc',
  ASC: 'asc',
};

export function descendingComparator(a, b, orderBy) {
  return a && b && b[orderBy].localeCompare(a[orderBy]);
}

export function getComparator(order, orderBy) {
  return order === 'desc'
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

export function stableSort(array, comparator) {
  const stabilizedThis = array.map((el, index) => [el, index]);
  stabilizedThis.sort((a, b) => {
    const order = comparator(a[0], b[0]);
    if (order !== 0) return order;
    return a[1] - b[1];
  });
  return stabilizedThis.map((el) => el[0]);
}
