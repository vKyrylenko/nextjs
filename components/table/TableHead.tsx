import { FunctionComponent } from 'react';
import MuiTableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';
import Checkbox from '@material-ui/core/Checkbox';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import { ITableHead } from './models';

export const TableHead: FunctionComponent<Partial<ITableHead>> = ({
  onSelectAllClick,
  order,
  orderBy,
  cells,
  numSelected,
  rowCount,
  onRequestSort,
}) => {
  const createSortHandler = (property) => () => {
    onRequestSort(property);
  };

  return (
    <MuiTableHead>
      <TableRow>
        <TableCell padding="checkbox">
          <Checkbox
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={rowCount > 0 && numSelected === rowCount}
            onChange={onSelectAllClick}
            inputProps={{ 'aria-label': 'select all desserts' }}
          />
        </TableCell>
        {cells.map((headCell) => (
          <TableCell
            key={headCell.name}
            align="left"
            style={headCell.width ? { width: `${headCell.width}%`, minWidth: `${headCell.width}%` } : undefined}
            sortDirection={orderBy === headCell.name ? order : false}
          >
            {headCell.isSortable ? (
              <TableSortLabel
                active={orderBy === headCell.name}
                direction={orderBy === headCell.name ? order : 'asc'}
                onClick={createSortHandler(headCell.name)}
              >
                {headCell.name}
              </TableSortLabel>
            ) : (
              headCell.name
            )}
          </TableCell>
        ))}
      </TableRow>
    </MuiTableHead>
  );
};
