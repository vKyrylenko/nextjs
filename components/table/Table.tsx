import React, { FunctionComponent, useState } from 'react';
import { Paper } from '@material-ui/core';
import TableContainer from '@material-ui/core/TableContainer';
import MuiTable from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import MuiTableRow from '@material-ui/core/TableRow';
import TablePagination from '@material-ui/core/TablePagination';
import TableCell from '@material-ui/core/TableCell';
import { TableRow } from './TableRow';
import { TableHead } from './TableHead';
import { TableToolBar } from './TableToolBar';
import { useStyles } from './Table.styles';
import { ITable } from './models';
import { getComparator, SortOrders, stableSort } from './utils';

const ROWS_PER_PAGE = 5;

export const Table: FunctionComponent<ITable> = ({
  data,
  defaultSortColumn,
  cells,
  onRemoveItem,
  onToggleAddItem,
  onSearch,
  onExportItems,
}) => {
  const classes = useStyles();
  const [order, setOrder] = useState(SortOrders.ASC);
  const [orderBy, setOrderBy] = useState(defaultSortColumn);
  const [selected, setSelected] = useState([]);
  const [page, setPage] = useState(0);

  const handleRequestSort = (property) => {
    const isAsc = orderBy === property && order === SortOrders.ASC;
    setOrder(isAsc ? SortOrders.DESC : SortOrders.ASC);
    if (orderBy !== property) {
      setOrderBy(property);
    }
  };

  const handleSelectAllClick = (event) => {
    if (event.target.checked) {
      const newSelecteds = data.map((n) => n.isbn);
      setSelected(newSelecteds);
      return;
    }
    setSelected([]);
  };

  const handleClick = (event, name) => {
    const selectedIndex = selected.indexOf(name);
    let newSelected = [];

    if (selectedIndex === -1) {
      newSelected = newSelected.concat(selected, name);
    } else if (selectedIndex === 0) {
      newSelected = newSelected.concat(selected.slice(1));
    } else if (selectedIndex === selected.length - 1) {
      newSelected = newSelected.concat(selected.slice(0, -1));
    } else if (selectedIndex > 0) {
      newSelected = newSelected.concat(selected.slice(0, selectedIndex), selected.slice(selectedIndex + 1));
    }

    setSelected(newSelected);
  };

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const onRemoveButtonClick = () => {
    onRemoveItem(selected);
    setSelected([]);
  };

  const onExportButtonClick = () => {
    onExportItems(selected);
    setSelected([]);
  };

  const onToggleAddButtonClick = () => {
    onToggleAddItem();
  };

  const isSelected = (name) => selected.indexOf(name) !== -1;

  return (
    <div className={classes.root}>
      <Paper className={classes.paper}>
        <TableToolBar
          numSelected={selected.length}
          onRemoveButtonClick={onRemoveButtonClick}
          onToggleAddButtonClick={onToggleAddButtonClick}
          onSearch={onSearch}
          onExportButtonClick={onExportButtonClick}
        />
        <TableContainer>
          <MuiTable stickyHeader={true} size="medium">
            <TableHead
              numSelected={selected.length}
              cells={cells}
              order={order}
              orderBy={orderBy}
              onSelectAllClick={handleSelectAllClick}
              onRequestSort={handleRequestSort}
              rowCount={data.length}
            />
            <TableBody>
              {stableSort(data, getComparator(order, orderBy))
                .slice(page * ROWS_PER_PAGE, page * ROWS_PER_PAGE + ROWS_PER_PAGE)
                .map((row) => {
                  const isItemSelected = isSelected(row.isbn);
                  return (
                    <TableRow
                      row={row}
                      isSelectable={true}
                      onClick={handleClick}
                      aria-checked={isItemSelected}
                      keyName={row.isbn}
                      key={row.isbn}
                      selected={isItemSelected}
                      cells={cells}
                    />
                  );
                })}
              {!data.length && (
                <MuiTableRow>
                  <TableCell colSpan={6}>No data</TableCell>
                </MuiTableRow>
              )}
            </TableBody>
          </MuiTable>
        </TableContainer>
        <TablePagination
          component="div"
          count={data.length}
          rowsPerPageOptions={[]}
          rowsPerPage={ROWS_PER_PAGE}
          labelRowsPerPage=""
          page={page}
          onChangePage={handleChangePage}
        />
      </Paper>
    </div>
  );
};
