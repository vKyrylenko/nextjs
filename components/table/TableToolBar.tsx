import React, { FocusEvent, FunctionComponent, useState } from 'react';
import { Toolbar } from '@material-ui/core';
import Tooltip from '@material-ui/core/Tooltip';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import DeleteIcon from '@material-ui/icons/Delete';
import AddIcon from '@material-ui/icons/Add';
import ImportExportIcon from '@material-ui/icons/ImportExport';
import TextField from '@material-ui/core/TextField';
import classNames from 'classnames';
import { ITableToolBar } from './models';
import { useToolbarStyles } from './Table.styles';

export const TableToolBar: FunctionComponent<ITableToolBar> = ({
  onRemoveButtonClick,
  onToggleAddButtonClick,
  numSelected,
  onSearch,
  onExportButtonClick,
}) => {
  const classes = useToolbarStyles();
  const [filter, setFilter] = useState('');

  const changeFilterHandler = (e: FocusEvent<HTMLInputElement>) => {
    setFilter(e.target.value);
    onSearch(e.target.value);
  };

  return (
    <Toolbar
      className={classNames(classes.root, {
        [classes.highlight]: numSelected > 0,
      })}
    >
      {numSelected > 0 ? (
        <Typography className={classes.title} color="inherit" variant="subtitle1" component="div">
          {numSelected} selected
        </Typography>
      ) : (
        <h3 className={classes.title}>Library</h3>
      )}

      {onSearch && !numSelected && <TextField defaultValue={filter} onBlur={changeFilterHandler} />}

      {!numSelected && onExportButtonClick ? (
        <Tooltip title="Export">
          <IconButton aria-label="export" onClick={onExportButtonClick}>
            <ImportExportIcon />
          </IconButton>
        </Tooltip>
      ) : null}
      {numSelected > 0 ? (
        <Tooltip title="Delete">
          <IconButton aria-label="delete" onClick={onRemoveButtonClick}>
            <DeleteIcon />
          </IconButton>
        </Tooltip>
      ) : null}
      <Tooltip title="Add">
        <IconButton aria-label="add" onClick={onToggleAddButtonClick}>
          <AddIcon />
        </IconButton>
      </Tooltip>
    </Toolbar>
  );
};
