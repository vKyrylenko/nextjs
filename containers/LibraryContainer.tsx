import React, { FunctionComponent, useEffect, useRef, useState } from 'react';
import { CSVLink } from 'react-csv';
import Link from 'next/link';
import { Table } from '../components/table/Table';
import { AuthorsCellRenderer } from '../components/AuthorsCellRenderer';
import { IBook, IMagazine } from '../schemas';
import { NewBookDialog } from '../components/NewBookDialog';

export const CELLS = [
  {
    name: 'title',
    isSortable: true,
    width: 17,
  },
  {
    name: 'authors',
    cellRenderer: (data, cellName) => {
      return <AuthorsCellRenderer authors={data[cellName]} />;
    },
    width: 18,
  },
  {
    name: 'description/publishedAt',
    cellRenderer: (data) => data.description || data.publishedAt,
    width: 55,
  },
  {
    name: 'view',
    cellRenderer: (data) => (
      <Link href={`/library/${data.entity}/${data.isbn}`}>
        <a>View</a>
      </Link>
    ),
    width: 10,
  },
];

const isPartlyIncludes = (item, field, search) => {
  return item[field].toString().toLowerCase().includes(search.toLowerCase());
};

interface ILibraryContainer {
  data: Array<IBook | IMagazine>;
}

export const LibraryContainer: FunctionComponent<ILibraryContainer> = ({ data }) => {
  const [library, setLibrary] = useState(data);
  const exportButton = useRef();
  const [newDialog, setNewDialog] = useState(false);
  const [exportFiles, setExportFiles] = useState(false);

  useEffect(() => {
    if (exportButton && exportButton.current && exportFiles) {
      // @ts-ignore
      exportButton.current.link.click();
      setExportFiles(false);
    }
  }, [exportButton, exportFiles]);

  const onRemoveItem = (ids: string[]) => {
    setLibrary((prevLibrary) =>
      prevLibrary.filter(
        ({ isbn }) =>
          !ids.some((id) => {
            return id === isbn;
          }),
      ),
    );
  };

  const onExportItems = (ids: string[]) => {
    setExportFiles(true);
  };

  const onToggleAddItem = () => {
    setNewDialog((prevState) => !prevState);
  };

  const onSaveItem = (item) => {
    setLibrary((prevState) => [...prevState, item]);
  };

  const onSearch = (search) => {
    if (!search) {
      setLibrary(data);
    }
    setLibrary((prevState) =>
      prevState.filter((item) => {
        return isPartlyIncludes(item, 'isbn', search) || isPartlyIncludes(item, 'authors', search);
      }),
    );
  };

  return (
    <>
      <Table
        data={library}
        defaultSortColumn="title"
        cells={CELLS}
        onRemoveItem={onRemoveItem}
        onExportItems={onExportItems}
        onToggleAddItem={onToggleAddItem}
        onSearch={onSearch}
      />
      {newDialog && <NewBookDialog onClose={onToggleAddItem} onSave={onSaveItem} />}

      <CSVLink style={{ display: 'none' }} ref={exportButton} data={library} filename="library.csv" />
    </>
  );
};
