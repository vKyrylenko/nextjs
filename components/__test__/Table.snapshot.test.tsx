import React from 'react';
import renderer from 'react-test-renderer';
import { Table } from '../table/Table';
import { CELLS } from '../../containers/LibraryContainer';

it('renders correctly', () => {
  const tree = renderer.create(<Table data={[]} cells={CELLS} defaultSortColumn="title" />).toJSON();
  expect(tree).toMatchSnapshot();
});
