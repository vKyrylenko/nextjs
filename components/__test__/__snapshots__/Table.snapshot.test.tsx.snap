// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-root-4 MuiToolbar-gutters"
    >
      <h3
        className="makeStyles-title-6"
      >
        Library
      </h3>
      <button
        aria-describedby={null}
        aria-label="export"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Export"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 3L5 6.99h3V14h2V6.99h3L9 3zm7 14.01V10h-2v7.01h-3L15 21l4-3.99h-3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-describedby={null}
        aria-label="add"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Add"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiTableContainer-root"
    >
      <table
        className="MuiTable-root MuiTable-stickyHeader"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-stickyHeader"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-10"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </span>
            </th>
            <th
              aria-sort="ascending"
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "minWidth": "17%",
                  "width": "17%",
                }
              }
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                title
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "minWidth": "18%",
                  "width": "18%",
                }
              }
            >
              authors
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "minWidth": "55%",
                  "width": "55%",
                }
              }
            >
              description/publishedAt
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "minWidth": "10%",
                  "width": "10%",
                }
              }
            >
              view
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              colSpan={6}
            >
              No data
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="MuiTablePagination-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          className="MuiTablePagination-spacer"
        />
        <p
          className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          0-0 of 0
        </p>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Previous page"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Next page"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
